packages/devices/origin-device-registry-irec-local-api/src/device/device.controller.ts

```diff
@@ -52,10 +57,36 @@ export class DeviceController {
     async getAll(): Promise<PublicDeviceDTO[]> {
         const devices = await this.deviceService.findAll();
 
-        return devices?.map((device) => plainToClass(PublicDeviceDTO, device));
+        return devices?.map((device) =>
+            plainToClass(PublicDeviceDTO, device, { excludeExtraneousValues: true })
+        );
+    }
+
+    @Get('/device-type')
+    @ApiResponse({
+        status: HttpStatus.OK,
+        type: [CodeNameDTO],
+        description: 'Returns all IREC fuels'
+    })
+    getFuels(): CodeNameDTO[] {
+        const deviceTypes = this.deviceService.getDeviceTypes();
+
+        return deviceTypes.map((deviceType) => plainToClass(CodeNameDTO, deviceType));
+    }
+
+    @Get('/fuel-type')
+    @ApiResponse({
+        status: HttpStatus.OK,
+        type: [CodeNameDTO],
+        description: 'Returns all IREC fuels types'
+    })
+    getFuelTypes(): CodeNameDTO[] {
+        const fuelTypes = this.deviceService.getFuelTypes();
+
+        return fuelTypes.map((fuelType) => plainToClass(CodeNameDTO, fuelType));
     }
 
-    @Get('/:id')
+    @Get('/device/:id')
```

Comment:
I've added "device" prefix here and for @Put('/device/:id') because otherwise it matches to GET /fuel-type and /device-type

Reply:
@kosecki123